p2-landing-page/js/app.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * 
+ * Manipulating the DOM exercise.
+ * Exercise programmatically builds navigation,
+ * scrolls to anchors from navigation,
+ * and highlights section in viewport upon scrolling.
+ * 
+ * Dependencies: None
+ * 
+ * JS Version: ES2015/ES6
+ * 
+ * JS Standard: ESlint
+ * 
+*/
+
+/**
+ * Comments should be present at the beginning of each procedure and class.
+ * Great to have comments before crucial code sections within the procedure.
+*/
+
+/**
+ * Define Global Variables
+ * 
+*/
+let sectionNavs = document.querySelectorAll("section");
+let NavTag = document.getElementById("navbar__list");
+let sectionNavPositions=[];
+let sectionNavLength = sectionNavs.length;
+
+// 1.build the nav get position info array
+sectionNavs.forEach((element,index)=>{
+    let sectionName=element.getAttribute("data-nav");
+    let scrollTarget=element.offsetTop;
+    let liTag=document.createElement('li');
+    liTag.setAttribute("class","nav"+index);
+    liTag.innerHTML=`<a onClick="scrollToSection(${scrollTarget})">${sectionName}</a>`;
+    NavTag.appendChild(liTag);
+})
+
+// 2.Scroll to anchor ID using scrollTO event
+function scrollToSection(sectionID) {
+	window.scrollTo({left:0, top:sectionID, behavior: 'smooth'});
+}
+
+// 3.Add class 'active' to section when near top of viewport
+document.addEventListener("scroll", () => {
+	sectionNavPositions = [];
+	sectionNavs.forEach((element) => sectionNavPositions.push(element.getBoundingClientRect().top+50));
+	let addIndex = sectionNavPositions.findIndex((element) => element > 0);
+	for (let i = 0; i < sectionNavLength; i++) {
+		if (addIndex === i) {
+			document.querySelector(".nav" + addIndex).classList.add("active");
+			document.querySelector(`#section${addIndex + 1}`).classList.add("current-active-class");
+		} else {
+			document.querySelector(".nav" + i).classList.remove("active");
+			document.querySelector(`#section${i + 1}`).removeAttribute("current-active-class");
+		}
+	}
+});
+
+    // findIndex can solve below 
```

Comment:
Are lines 61-70 still necessary code?

Reply:
Thanks! 
That's some commented code i left there, that time i was trying to do the things function findIndex did. then i search and find findIndex, but dont want to delete them since i spent some time making it. if in real project i know i need to delete and keep it clean, but for learning project i just put there as a reminder. realize it or find something already there can do the work.